File: android/src/com/panyam/mango/templates/utils/Pair.java
package com.panyam.mango.templates.utils;

/**
 * A generic pair object.
 * 
 * @author Sri Panyam
 *
 * @param <A>    Type of the first object.
 * @param <B>    Type of the second object.
 */
public class Pair<A, B> 
{
    public A first;
    public B second;
    
    public Pair(A a, B b)
    {
        first = a;
        second = b;
    }
}

File: iphone/Classes/utils/Pair.m
//
//  Pair.m
//  DjangoTemplates
//
//  Created by Sri Panyam on 16/02/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#include "Pair.h"

@implementation Pair

@synthesize first;
@synthesize second;

- (id) initWithValues:(id)f second:(id)s
{
    if (self = [super init])
    {
        self.first = [f retain];
        self.second = [s retain];
    }
    return self;
}

-(void) dealloc
{
    [first release];
    [second release];
	[super dealloc];
}

@end
